File: algorithms/392. Is Subsequence/solution.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        i = 0
        for c in s:
            while i < len(t) and t[i] != c:
                i += 1
            i += 1
        return i <= len(t)

File: algorithms/392. Is Subsequence/solution.cc
class Solution {
public:
    bool isSubsequence(string s, string t) {
        vector<vector<int>> indices(26);
        for ( int i = 0; i < t.size(); ++i ) {
            indices[t[i] - 'a'].push_back(i);
        }
        for ( int pre = -1, i = 0; i < s.size(); ++i ) {
            auto & idx = indices[s[i] - 'a'];
            auto iter = upper_bound(idx.begin(), idx.end(), pre);
            if ( iter == idx.end() ) {
                return false;
            }
            pre = *iter;
        }
        return true;
    }
};
